### DISCUSSION TITLE
Work out how to open password protected .xlsb files via R

### DISCUSSION THREAD

**Discussion by adamkucharski** [WHO-Collaboratory/collaboratory-epipipeline-community#20] [Ideas]
WHO Collaboratory Hackathon Idea.

**Comment by ntncmch** [comment]
Hi Martina (@martinamcm), I think you were interested in this subject, weren't you? If so, Patrick (@patrickbarks) has found a solution that he can direct you to. Cheers.

**Reply by patrickbarks** [reply]
Hi Martina, we came across a Python package ([msoffcrypto-tool](https://pypi.org/project/msoffcrypto-tool/)) that can decrypt password-protected xlsx or xlsb files, with a user-supplied password, so we've incorporated this into a quick R package [rpxl](https://github.com/epicentre-msf/rpxl) to read password-protected Excel files directly into R. It relies on the R package [reticulate](https://rstudio.github.io/reticulate/index.html) to interface with Python. The initial setup of the Python environment and dependencies can be a bit difficult, but we've got it to work on at least one of each machine type so far (Mac/PC/linux). If you have any suggestions, questions, or issues, feel free to reach out!

**Comment by martinamcm** [comment]
Hi Anton and Paul,Very sorry for the delay in replying - busy week! Thanks so much for reaching out and for providing such a nice solution - this is super useful! I’ll test it out my side and see how it goes, hopefully I can get it working. Thanks again, Martina Sent from my iPhoneOn 5 Oct 2023, at 3:24 PM, Patrick Barks ***@***.***> wrote:﻿
Hi Martina, we came across a Python package (msoffcrypto-tool) that can decrypt password-protected xlsx or xlsb files, with a user-supplied password, so we've incorporated this into a quick R package rpxl to read password-protected Excel files directly into R. It relies on the R package reticulate to interface with Python. The initial setup of the Python environment and dependencies can be a bit difficult, but we've got it to work on at least one of each machine type so far (Mac/PC/linux). If you have any suggestions, questions, or issues, feel free to reach out!
